### PR TITLE
[ui-leftnav] Remove "Edit List" tab from editor dropdown in left navbar

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Earlier, the `Edit List...` tab was generation no UX value and clicking on it was redirecting to gethue docs connector page for non-connector setup scenarios.
- This change removes the `Edit List...` tab from the editor dropdown in the left navigation. The plan is to make the editor dropdown section clean and to only show the configured editors.

## How was this patch tested?

- Manually

### CODE REVIEW

File: desktop/core/src/desktop/static/desktop/locales/en/translation.json [hunk 1]
```diff
@@ -138,7 +138,6 @@
   "Drop a SQL file here": "Drop a SQL file here",
   "Drop iPython/Zeppelin notebooks here": "Drop iPython/Zeppelin notebooks here",
   "Edit": "Edit",
-  "Edit list...": "Edit list...",
```
Comment (Harshg999) [inline desktop/core/src/desktop/static/desktop/locales/en/translation.json:141]: @JohanAhlen @bjornalm - Does it makes sense to remove it from here explicitly or we should regenerate this translation json again? If its the latter, then how can we generate it (maybe as a separate commit)?

Comment (JohanAhlen) [reply desktop/core/src/desktop/static/desktop/locales/en/translation.json:141]: Yeah, good to remove here if it's no longer used.

**Review by JohanAhlen — APPROVED** [review]
Nice!

**Review by agl29 — APPROVED** [review]
LGTM!

**Pull request merged**